draft-ietf-masque-connect-udp.md

```diff
@@ -403,18 +442,90 @@ Reference:
 : This document.
 
 
-## Datagram Format Type {#iana-format-type}
+--- back
+
+# Example Extensions
 
-This document will request IANA to register UDP_PAYLOAD in the "HTTP Datagram
-Format Types" registry established by {{HTTP-DGRAM}}.
+Extensions can define new semantics for the payload of HTTP Datagrams. The
+extension can then have an endpoint pick an available context ID using the
+allocation service ({{context-id}}) and register that context ID with their
+peer.
 
-|    Type     |   Value   | Specification |
-|:------------|:----------|:--------------|
-| UDP_PAYLOAD | 0xff6f00  | This Document |
-{: #iana-format-type-table title="Registered Datagram Format Type"}
 
+## Registering Contexts with Headers
+
+Extensions can define a new HTTP header to register a context ID with the peer
+endpoint.
+
+As an example, take an extension that conveys the time at which a UDP
+packet was received. The extension would first define the format of its HTTP
+Datagram Payload field:
+
+~~~
+UDP with Timestamp HTTP Datagrams {
+  Context ID (i),
+  Timestamp (64),
+  UDP Payload (..),
+}
+~~~
+{: #ex-dgram title="Example: Format of UDP Payload with Timestamp"}
+
+The extension would also define a new HTTP header (Example-UDP-Timestamps) that
+includes a context ID value. Servers that understand this new HTTP header would
+be able to consequently handle and parse datagrams on the context ID, while all
+other servers would silently drop the datagrams. Clients also indicate their
+support for receiving on this Context ID by registering it.
```

Comment:
1. I personally don't think there's value in an acknowledgement, but I think that's something we'll need to hash out with the WG when we discuss extension composability and registration
2. In general, yes. Individual extensions can prohibit it though
3. Not in this particular example, but an extension can absolutely do that

At a high level, this appendix is most likely not going to survive publication, let's see what it looks like once we've figured out registration and composability with the WG

Reply:
I'm not really comfortable with that, because it means we'll just rediscover these points.

If we don't want to fix up this appendix, let's move it into a wiki or whatever.

Reply:
I've added an explicit note stating that this appendix will be removed before WGLC

Reply:
This would not be my preference. If the point of the appendix is to help make this clearer, then having it be confusing does not achieve that.

Reply:
Fair enough, I've added a paragraph that answers your three questions above